python/lsst/sconsUtils/state.py

```diff
@@ -128,6 +130,11 @@ def _initEnvironment():
                 ourEnv[varname] = os.environ[varname]
                 ourEnv[k] = os.environ[k]
 
+    # add <build root>/ups directory to the configuration search path
+    # this allows the .cfg file for the package being built to be found without
+    # requiring <product name>_DIR to be in the env
+    cfgPath.append(os.path.join(SCons.Script.Dir('#').abspath, 'ups'))
```

Comment:
I personally have 91 environment variables with `_DIR` in the name. I didn't realise they all got searched. Shouldn't `scons` only be looking for `_DIR` variables associated with entries in the table file of the application being built? I guess EUPS can't use something like `_EDIR` instead? (it would break assumptions in the new lsst.utils routine for finding the product directory).